09_Loading_Elf/02_Loading_And_Running.md

```diff
@@ -14,14 +14,48 @@ In the previous chapter we looked at the details of loading program headers, but
-- Find all the loadable program headers for the ELF, we'll need those in a moment.
+- Find all the loadable program headers for the ELF, we'll need those in a moment (They are the headers with type `PT_LOAD`).
 - Create a new address space for the program to live in. This usually involves creating a new process with a new VMM instance, but the specifics will vary depending on your design. Don't forget to keep the kernel mappings in the higher half!
 - Copy the loadable program headers into this new address space. Take care when writing this code, as the program headers may not be page-aligned:. Don't forget to zero the extra bytes between `memsz` and `filesz`.
 - Once loaded, set the appropriate permission on the memory each program header lives in: the write, execute (or no-execute) and user flags.
 - Now we'll need to create a new thread to act as the main thread for this program, and set its entry point to the `e_entry` field in the ELF header. This field is the start function of the program. You'll also need to create a stack in the memory space of this program for the thread to use, if this wasnt already done as part of our thread creation.
 
 If all of the above are done,  then the program is ready to run! We now should be able to enqueue the main thread in the scheduler and let it run.
 
+### What We Verify?
+
+When veryfyig an ELF file there are few things we need to check in order to decide if an executable is valid, the field to validate are in different headers, some of them are in the `e_ident` header, and they are the following:
+
+* The first thing we want to check is the Magic number, this is the `ELFMAG` part. It is expected to be the following values: `0x7f, 'E', 'L', 'F'`. Bytes 0 to 3.
+* We need to check that the file class match with the one we are supporting. There are two possible classes: 64 and 32. Thi is byte 4
+* The data field indicates the bit numbering convetion, again this depends on the architecture used. It can be three values: None (0), LSB (1) and MSB (2). For example x86_64 architecture value is 1. This field is in the byte 5.
+* The version field, byte 6,  to be a valid elf it has to be set to 1 (EVCURRENT).
+* The OS Abi and Abi version they  identify the operating system together with the ABI to which the object is targeted and the version of the ABI to which the object is targeted, for now we can ignore them, the should be 0.
+
+Then from the other fields that needs validation (that area not in the `e_ident` field) are:
+
+* `e_type`: they identify the type of elf, for our purpose the one to be considered valid this value should be 2 that indicates an Executable File (ET_EXEC) there are other values that in the future we could support, but they require more work to be done.
```

Comment:
they -> this.
It might be nice to mention that the other value to look out for is ET_DYN, which is for position-independent code (executables or shared objects). It does require a more complex loader so we wont cover it now, but it is worth keeping in mind.